### PR TITLE
[567] Phrase selection repetition bug fix

### LINKED ISSUE #567: Vocable iOS - iOS 14 Erroneously Selecting Phrases Repeatedly
https://user-images.githubusercontent.com/143916/163464439-53d78fd1-adbc-43ea-9978-bf02f7211dd9.MP4

As seen in Robbie's video, iOS 14 is seeing infinite selections as long as the cursor is positioned over a phrase cell.

**To Reproduce**
1. Gaze the cursor over a phrase cell on an iOS 14 device
2. Observe as the phrase is spoken ad infinitum 

**Expected behavior**
1. One selection per gaze event

**Actual behavior**
See video

**Device Information**
 - Device: Any
 - OS: 14.x

### PR DESCRIPTION
**Pull request by Clstroud**
Closes #567 

# Description of Work
* Fixes the underlying issue causing the repetition
* Makes some heavily used UI elements update lazily to reduce overhead
* Moves infinite scrolling datasource preparation to a background queue for smoother updates

### CODE REVIEW

File: Vocable/Features/Root/CategoriesCarouselViewController.swift [hunk 1]
```diff
@@ -42,18 +44,15 @@ import Combine
...
 
-    private lazy var dataSourceProxy = CarouselCollectionViewDataSourceProxy<String, NSManagedObjectID>(collectionView: collectionView!) { [weak self] (collectionView, indexPath, _) -> UICollectionViewCell? in
-        guard let self = self else { return nil }
-        let category = self.frc.object(at: indexPath)
-        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: CategoryItemCollectionViewCell.reuseIdentifier, for: indexPath) as! CategoryItemCollectionViewCell
-        cell.setup(title: category.name!)
-        cell.accessibilityIdentifier = ["category_title_cell", category.identifier].compacted().joined(separator: "_")
-        return cell
-    }
+    private var dataSourceProxy: DataSource!
```
Comment (jsmorgan42) [inline Vocable/Features/Root/CategoriesCarouselViewController.swift:47]: Could we not instead just make this a private let that's instantiated on initialization of the view controller?

File: Vocable/Common/CarouselCollectionViewDataSourceProxy.swift [hunk 2]
```diff
@@ -10,36 +10,44 @@ import UIKit
 
 class CarouselCollectionViewDataSourceProxy<SectionIdentifier: Hashable, ItemIdentifier: Hashable>: NSObject {
 
+    private typealias Snapshot = NSDiffableDataSourceSnapshot<SectionIdentifier, ItemIdentifier>
+    private typealias Impl = UICollectionViewDiffableDataSource<ContentWrapper<SectionIdentifier>, ContentWrapper<ItemIdentifier>>
```
Comment (jsmorgan42) [inline Vocable/Common/CarouselCollectionViewDataSourceProxy.swift]: I know this was a pre-existing variable name but I wasn't sure what `Impl` is supposed to be short for in this context 🤔

Comment (jsmorgan42) [reply Vocable/Common/CarouselCollectionViewDataSourceProxy.swift]: Maybe change to `Datasource`

**Review by jsmorgan42 — APPROVED** [review]
💯 💰 🚢

**Review by jsmorgan42 — CHANGES_REQUESTED** [review]
Looks like once you start switching between categories, the phrases are not displayed anymore, instead it only shows a blank screen

**Review by jsmorgan42 — APPROVED** [review]

**Comment by rowildcard** [comment]
Test verified on iPhone X, iOS 14.2
Verified:
Selecting preset phrases in each category. ✅ 
Selecting preset phrases on second page of a category. ✅ 
Add custom category and phrase; select custom phrase. ✅

**Pull request merged**